portfolio/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/auth")
+public class AuthServlet extends HttpServlet {
+
+  public static final String AUTH_URL_REDIRECT = "/";
+
+  public Gson gson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {    
+    response.setContentType("application/json;");
+
+    JsonObject responseObject = new JsonObject();
+
+    UserService userService = UserServiceFactory.getUserService();
+    String loginUrl = userService.createLoginURL(AUTH_URL_REDIRECT);
+
+    responseObject.addProperty("loginUrl", loginUrl);
+
+    if (userService.isUserLoggedIn()) {
+      responseObject.addProperty("authorized", true);
+
+      User currentUser = userService.getCurrentUser();
+      UserData userData = new UserData(currentUser);
```

Comment:
The variable name `currentUser` doesn't tell the reader any information that they couldn't have figured out from `userService.getCurrentUser()`. Since `currentUser` is only used once, you should condense these two lines into one by skipping the step where you store `userService.getCurrentUser()` in a variable.

```suggestion
      UserData userData = new UserData(userService.getCurrentUser());
```

This is a bit more of an art than a science because you don't want to have just one super-long line of code. If you take this argument to its logical conclusion you should combine all of the variables into one line like
`responseObject.add("user", gson.toJsonTree(new UserData(userService.getCurrentUser())));` which is pretty difficult for me to read because there are a ton of parentheses to keep track of.

Reply:
Yeah I guess it's about kind of reading the line "out loud" in your head. When it starts to sound too jumbled is I think when it's too much for one line. Reading out "get the current user from the user service and create user data" is doable.